### PR TITLE
restore the maxPayload option

### PR DESCRIPTION
**Pull request by darrachequesne**
It seems the option was present in earlier versions: https://github.com/uWebSockets/uWebSockets/blob/v0.8.0/src/Server.h#L104

Related: https://github.com/uWebSockets/uWebSockets/issues/299

### CODE REVIEW

**Comment by alexhultman** [comment]
Oh wow, I had a pending review comment for days and I didn't understand it was private to me. Anyways here it is in comment form:

> Well it is partially incomplete, there are more paths that construct a websocket. But anyways - the maxPayload per websocket should be stored per group. So Hub takes the option and passes it to all Groups created from this Hub. Websockets then read the maxPayload from their group. Also, the default of 100mb should be 16mb which is the current value.

Btw, is Socket..IO 2.0 coming out anytime soon? What's holding it up?

File: src/WebSocket.h [hunk 1]
```diff
@@ -32,7 +33,8 @@ struct WIN32_EXPORT WebSocket : uS::Socket, WebSocketState<isServer> {
     using uS::Socket::closeSocket;
 
     static bool refusePayloadLength(uint64_t length, WebSocketState<isServer> *webSocketState) {
-        return length > 16777216;
+        WebSocket<isServer> *webSocket = static_cast<WebSocket<isServer> *>(webSocketState);
+        return length > webSocket->maxPayload;
```
Comment (alexhultman) [inline src/WebSocket.h]: This would then be something like Group<isServer>::from(webSocket)->maxPayload

File: src/WebSocket.h [hunk 2]
```diff
@@ -22,8 +22,9 @@ struct WIN32_EXPORT WebSocket : uS::Socket, WebSocketState<isServer> {
         COMPRESSED_FRAME
     } compressionStatus;
     unsigned char controlTipLength = 0, hasOutstandingPong = false;
+    unsigned int maxPayload;
```
Comment (alexhultman) [inline src/WebSocket.h]: This gets moved to Group

File: src/Hub.h [hunk 3]
```diff
@@ -43,8 +44,8 @@ struct WIN32_EXPORT Hub : private uS::Node, public Group<SERVER>, public Group<C
     void connect(std::string uri, void *user = nullptr, std::map<std::string, std::string> extraHeaders = {}, int timeoutMs = 5000, Group<CLIENT> *eh = nullptr);
     void upgrade(uv_os_sock_t fd, const char *secKey, SSL *ssl, const char *extensions, size_t extensionsLength, const char *subprotocol, size_t subprotocolLength, Group<SERVER> *serverGroup = nullptr);
 
-    Hub(int extensionOptions = 0, bool useDefaultLoop = false) : uS::Node(LARGE_BUFFER_SIZE, WebSocketProtocol<SERVER, WebSocket<SERVER>>::CONSUME_PRE_PADDING, WebSocketProtocol<SERVER, WebSocket<SERVER>>::CONSUME_POST_PADDING, useDefaultLoop),
-                                             Group<SERVER>(extensionOptions, this, nodeData), Group<CLIENT>(0, this, nodeData) {
+    Hub(int extensionOptions = 0, bool useDefaultLoop = false, unsigned int maxPayload = 1048576) : uS::Node(LARGE_BUFFER_SIZE, WebSocketProtocol<SERVER, WebSocket<SERVER>>::CONSUME_PRE_PADDING, WebSocketProtocol<SERVER, WebSocket<SERVER>>::CONSUME_POST_PADDING, useDefaultLoop),
```
Comment (alexhultman) [inline src/Hub.h]: This is fine, but should be 16mb by default

**Comment by darrachequesne** [comment]
@alexhultman thanks for the merge. Version 2.0 will come very soon (:tm:), but I'd like to include that PR.

**Comment by alexhultman** [comment]
Ah, the "Coming soon, Inc", patent pending :wink: Thanks for the PR, if you approach release you can ping me 1 week prior and I can release a version with this. I will need to add some more modifications to the limit of permessage-deflate but I can do that just prior to release.

**Comment by darrachequesne** [comment]
@alexhultman there's still a few issues to tackle, but I'll try to release 2.0 by next week.

**Comment by alexhultman** [comment]
Alright I have finished 0.14.3 now, will publish sometime soon. It should be all set

**Pull request merged**